scripts/go-helper.sh

```diff
@@ -8,41 +8,68 @@ set -euo pipefail
 check_fmt() {
   echo "==> Checking code formatting..."
 
-  declare -a malformed=()
-  IFS=" " read -r -a files <<< "$(tr '\n' ' ' <<< "$@")"
+  declare -a malformed_imports=()
+  declare -a malformed_formatting=()
+  local failed=0
+  IFS=" " read -r -a files <<<"$(tr '\n' ' ' <<<"$@")"
   if [ -n "${files+set}" ] && [[ "${files[0]}" != "" ]]; then
     echo "--> Checking changed files..."
     for file in "${files[@]}"; do
       if [ ! -f "$file" ]; then
-        echo "--> $file no longer exists ⚠"
+        echo "----> $file no longer exists ⚠"
         continue
       fi
 
-      if echo "$file" | grep -v pb.go | grep -v vendor > /dev/null; then
+      if echo "$file" | grep -v pb.go | grep -v vendor >/dev/null; then
         local output
+        if ! output=$(gosimports -d "$file") || [ "$output" != "" ]; then
+          echo "----> ${file} ✖ (gosimports)"
+          malformed_imports+=("$file")
+          failed=1
+        fi
         if ! output=$(gofumpt -l "$file") || [ "$output" != "" ]; then
-          echo "--> ${file} ✖"
-          malformed+=("$file")
+          echo "----> ${file} ✖ (gofumpt)"
+          malformed_formatting+=("$file")
+          failed=1
+        fi
+
+        if [ "$failed" == 1 ]; then
           continue
         fi
       fi
 
-      echo "--> ${file} ✔"
+      echo "----> ${file} ✔"
     done
   else
     echo "--> Checking all files..."
-    IFS=" " read -r -a malformed <<< "$(find . -name '*.go' | grep -v pb.go | grep -v vendor | xargs gofumpt -l)"
+    IFS=" " read -r -a malformed_imports <<<"$(find . -name '*.go' | grep -v pb.go | grep -v vendor | xargs gosimports -l | tr '\n' ' ')"
+    IFS=" " read -r -a malformed_formatting <<<"$(find . -name '*.go' | grep -v pb.go | grep -v vendor | xargs gofumpt -l)"
+  fi
+
+  if [ "${#malformed_imports[@]}" -ne 0 ] && [ -n "${malformed_imports[0]}" ]; then
+    failed=1
+    echo "--> The following files need to be reformatted with gosimports"
+    printf '%s\n' "${malformed_imports[*]}"
+    #printf '%s\n' "${malformed_imports[@]}"
```

Comment:
Safe to delete the comment?

Reply:
It sure is! But then we get to re-review all the things!